compute/data_spark_version.go

```diff
@@ -0,0 +1,94 @@
+package compute
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceClusterSparkVersions() *schema.Resource {
+	return &schema.Resource{
+		Read: DataSourceClusterSparkVersionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true},
+			"latest": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"long_term_support": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"ml": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"gpu": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"scala": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "2.12",
+			},
+			"genomics": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+// DataSourceClusterSparkVersionsRead performs the Databricks runtime lookup.
+func DataSourceClusterSparkVersionsRead(d *schema.ResourceData, m interface{}) error {
+	sparkVersions, err := NewClustersAPI(m).ListSparkVersions()
+	if err != nil {
+		return err
+	}
+	latest := d.Get("latest").(bool)
```

Comment:
can you change this logic the way that we have `Latest` method on `SparkVersionList` and can use that programmatically in tests as well? You'll have to rewrite data source by using `DataSourceGroup` as reference.

```
type SparkVersionRequest struct {
	LongTermSupport bool `json:"long_term_support,omitempty"`
	ML bool `json:"ml,omitempty"`
	GPU bool `json:"gpu,omitempty"`
	Scala string `json:"scala,omitempty"`
	ApacheSpark string `json:"apache_spark,omitempty"`
}

//.. add method on SparkVersionList, so that we can use it in integration tests if needed

func (svl SparkVersionList) Latest(svr SparkVersionRequest) (string, error) {
	//..
}
```